src/api/client.ts

```diff
@@ -1,28 +1,51 @@
-import axios, { AxiosRequestConfig, AxiosInstance } from 'axios';
-
-import { config } from './config';
-
-const api = axios.create({
-  baseURL: config.baseURL,
-  headers: { Accept: 'application/json', 'Content-Type': 'application/json' },
-});
-
-export const setRequestInterceptor = (appInstance: AxiosInstance) =>
-  appInstance.interceptors.request.use(
-    async (config: AxiosRequestConfig) => {
-      //TODO: still no idea what here need improvements
-      return config;
-    },
-    error => Promise.reject(error),
-  );
-export const setResponseInterceptor = (appInstance: AxiosInstance) =>
-  appInstance.interceptors.response.use(
-    response => {
-      //TODO: do something with the response data
-      return response.data;
-    },
-    error => Promise.reject(error),
+import { HttpError } from 'app/../error/AppErrors';
+import logger from 'app/../logger';
+import axios, { AxiosRequestConfig, AxiosError, AxiosResponse } from 'axios';
+
+export enum GeneralHttpError {
+  NO_RESPONSE = 'No response',
+  NO_MESSAGE = 'No message',
+}
+
+const createHttpClient = (baseUrl: string) => {
+  const httpClient = axios.create({
+    baseURL: baseUrl,
+    headers: { 'Content-Type': 'application/json' },
+  });
+
+  const onRequest = (request: AxiosRequestConfig) => {
+    logger.info('http', `--> ${request.method?.toUpperCase()} ${request.baseURL}${request.url}`);
+
+    return request;
+  };
+
+  const onResponse = (response: AxiosResponse<any>): AxiosResponse<any> => {
+    logger.info('http', `<-- ${response.status} ${response.config.baseURL}${response.config.url}`);
+
+    return response;
+  };
+
+  const onError = (error: AxiosError) => {
+    if (error.response) {
+      logger.error('http', `<-- ${error.response.status} ${error.config.baseURL}${error.config.url}`);
+    }
+
+    if (!error?.response) {
+      throw new HttpError(`Request to ${baseUrl} failed. Details: No response`);
+    }
+
+    const message = error.response?.data.msg || 'No message';
+
+    throw new HttpError(`Request to ${error.config.url} failed. Details: ${message}`);
```

Comment:
I think that just message should be thrown here

Reply:
I made it for debugging and logging purposes - I think it'd be better to see in console/Sentry
```
Request to /verify-email/ failed. Details: Bad request: {'new_email': [ErrorDetail(string='Enter a valid email address.', code='invalid')]}
```

instead of
```
Bad request: {'new_email': [ErrorDetail(string='Enter a valid email address.', code='invalid')]}
```

Reply:
but if you add it here you would have to add it also to translations which I don't think is a good thing

Reply:
But why do you want to translate error messages? I really believe the messages like this are only for developers and we shouldn't display them explicitly for users. I know we do it right now, but it's a mistake - that's the reason why our users see messages "connection closed", "txn-mempool-conflict (code 18)" and so on. 

I think we should stop treat the `error.message` as a copy.

Reply:
ok, make sense, we can always use `includes` for mapping error